packages/hoppscotch-sh-admin/src/composables/useConfigHandler.ts

```diff
@@ -149,74 +152,45 @@ export function useConfigHandler(updatedConfigs?: ServerConfigs) {
-      enabled: updatedConfigs?.providers.google.enabled,
-      fields: updatedConfigs?.providers.google.fields,
-    },
-    {
-      config: GITHUB_CONFIGS,
-      enabled: updatedConfigs?.providers.github.enabled,
-      fields: updatedConfigs?.providers.github.fields,
-    },
-    {
-      config: MICROSOFT_CONFIGS,
-      enabled: updatedConfigs?.providers.microsoft.enabled,
-      fields: updatedConfigs?.providers.microsoft.fields,
-    },
-    {
-      config: MAIL_CONFIGS,
-      enabled: updatedConfigs?.mailConfigs.enabled,
-      fields: updatedConfigs?.mailConfigs.fields,
-    },
-  ];
+  // Transforming the working configs back into the format required by the mutations
+  const transformInfraConfigs = () => {
+    const updatedWorkingConfigs: ConfigTransform[] = [
+      {
+        config: GOOGLE_CONFIGS,
+        enabled: updatedConfigs?.providers.google.enabled,
+        fields: updatedConfigs?.providers.google.fields,
+      },
+      {
+        config: GITHUB_CONFIGS,
+        enabled: updatedConfigs?.providers.github.enabled,
+        fields: updatedConfigs?.providers.github.fields,
+      },
+      {
+        config: MICROSOFT_CONFIGS,
+        enabled: updatedConfigs?.providers.microsoft.enabled,
+        fields: updatedConfigs?.providers.microsoft.fields,
+      },
+      {
+        config: MAIL_CONFIGS,
+        enabled: updatedConfigs?.mailConfigs.enabled,
+        fields: updatedConfigs?.mailConfigs.fields,
+      },
+    ];
 
-  // Push or filter the configs based on the enabled condition
-  const transformInfraConfigs = (workingConfigs: ConfigTransform[]) => {
-    let newConfigs: UpdatedConfigs[] = [];
+    const transformedConfigs: UpdatedConfigs[] = [];
 
-    workingConfigs.forEach(({ config, enabled, fields }) => {
+    updatedWorkingConfigs.forEach(({ config, enabled, fields }) => {
       config.forEach(({ name, key }) => {
         if (enabled && fields) {
           const value =
             typeof fields === 'string' ? fields : String(fields[key]);
-          newConfigs.push({ name, value });
+          transformedConfigs.push({ name, value });
         }
       });
     });
-    return newConfigs;
-  };
-
-  // Transforming the working configs back into the format required by the mutations
-  const updatedInfraConfigs = () =>
-    updatedConfigs ? transformInfraConfigs(workingUpdatedConfigs) : [];
 
-  // Updated allowed auth providers
-  const updatedAllowedAuthProviders = [
-    {
-      provider: 'GOOGLE',
-      status: updatedConfigs?.providers.google.enabled ? 'ENABLE' : 'DISABLE',
-    },
-    {
-      provider: 'MICROSOFT',
-      status: updatedConfigs?.providers.microsoft.enabled
-        ? 'ENABLE'
-        : 'DISABLE',
-    },
-    {
-      provider: 'GITHUB',
-      status: updatedConfigs?.providers.github.enabled ? 'ENABLE' : 'DISABLE',
-    },
-    {
-      provider: 'EMAIL',
-      status: updatedConfigs?.mailConfigs.enabled ? 'ENABLE' : 'DISABLE',
-    },
-  ];
+    return transformedConfigs;
```

Comment:
We could rewrite this in a `reduce` based approach as below. Please feel free to push if you'd like it.

```ts
    return updatedWorkingConfigs.reduce(
      (transformedConfigs: UpdatedConfigs[], { config, enabled, fields }) => {
        if (enabled && fields) {
          config.forEach(({ name, key }) => {
            const value =
              typeof fields === 'string' ? fields : String(fields[key]);
            transformedConfigs.push({ name, value });
          });
        }
        return transformedConfigs;
      },
      []
    );
```